packages/cheerio-crawler/src/internals/cheerio-crawler.ts

```diff
@@ -1062,23 +1065,39 @@ export class CheerioCrawler<JSONData = unknown> extends BasicCrawler<
 interface EnqueueLinksInternalOptions {
     options?: CheerioCrawlerEnqueueLinksOptions;
     $: CheerioRoot | null;
-    requestQueue?: RequestQueue;
-    defaultBaseUrl?: string;
+    requestQueue: RequestQueue;
+    originalRequestUrlOrigin?: string;
+    finalRequestUrlOrigin?: string;
 }
 
 /** @internal */
-export async function cheerioCrawlerEnqueueLinks({ options, $, requestQueue, defaultBaseUrl }: EnqueueLinksInternalOptions) {
+export async function cheerioCrawlerEnqueueLinks({ options, $, requestQueue, originalRequestUrlOrigin, finalRequestUrlOrigin }: EnqueueLinksInternalOptions) {
     if (!$) {
         throw new Error('Cannot enqueue links because the DOM is not available.');
     }
 
-    const baseUrl = options?.baseUrl ?? defaultBaseUrl;
+    let baseUrl: string | undefined;
+
+    if (options?.baseUrl) {
+        // User provided base url takes priority
+        baseUrl = options.baseUrl;
+    } else if (options?.strategy === EnqueueStrategy.All) {
+        // We can assume users want to go off the domain in this case
+        baseUrl = finalRequestUrlOrigin ?? originalRequestUrlOrigin;
+    } else if (originalRequestUrlOrigin === finalRequestUrlOrigin) {
+        // If the domains are the same, we can use the original domain
+        baseUrl = originalRequestUrlOrigin;
+    }
```

Comment:
the code looks almost identical to the one in browser crawler, can we refactor this out? i guess this version would work for the browser crawler too, right?